src/forms/renderers/Overrides/material/controls/MaterialDateControl.tsx

```diff
@@ -0,0 +1,150 @@
+
+    const intl = useIntl();
+    const popupState = usePopupState({
+        variant: 'popover',
+        popupId: `date-picker-${id}`,
+    });
+
+    const formatDate = (value: Date) => {
+        try {
+            return format(value, Patterns.date);
+        } catch (e: unknown) {
+            return INVALID_DATE;
+        }
+    };
+
+    const onChange = (value: any, keyboardInput?: string | undefined) => {
+        if (value) {
+            const formattedValue = formatDate(value);
+            if (formattedValue && formattedValue !== INVALID_DATE) {
+                return handleChange(path, formattedValue);
+            }
+        }
+
+        // Default to setting to what user typed
+        //  This is a super backup as with the Date Fn adapter
+        //  it never fell through to this... but wanted to be safe
+        return handleChange(path, keyboardInput);
+    };
+
+    return (
+        <Hidden xsUp={!visible}>
+            <Stack
+                sx={{
+                    alignItems: 'top',
+                }}
+                direction="row"
+            >
+                <MaterialInputControl input={MuiInputText} {...props} />
+                <Box sx={{ paddingTop: 2 }}>
+                    <IconButton
+                        aria-label={intl.formatMessage(
+                            {
+                                id: 'datePicker.button.ariaLabel',
+                            },
+                            {
+                                label,
+                            }
+                        )}
+                        disabled={!enabled}
+                        {...bindTrigger(popupState)}
+                    >
+                        <EventIcon />
+                    </IconButton>
+                </Box>
+
+                <Popover
+                    {...bindPopover(popupState)}
+                    anchorOrigin={{
+                        vertical: 'center',
+                        horizontal: 'left',
+                    }}
+                    transformOrigin={{
+                        vertical: 'center',
+                        horizontal: 'right',
+                    }}
+                >
+                    <LocalizationProvider dateAdapter={AdapterDateFns}>
+                        <StaticDatePicker
+                            ignoreInvalidInputs
+                            disableMaskedInput
+                            displayStaticWrapperAs="desktop"
+                            openTo="day"
+                            disabled={!enabled}
+                            value={data}
+                            onChange={onChange}
+                            onAccept={popupState.close}
+                            closeOnSelect={true}
+                            // We don't need an input
+                            // eslint-disable-next-line react/jsx-no-useless-fragment
+                            renderInput={() => <></>}
```

Comment:
>renderInput={() => null}

Would this work in this instance?

Reply:
Nope - they want a ReactNode and do not allow `null` here